src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -39,6 +39,11 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
   }
 
   private def createVertex(workspace: Vertex, workspaceNamespace: String, workspaceName: String, entity: Entity, txn: RawlsTransaction): Vertex = txn withGraph { db =>
+    // check for illegal dot characters
+    validateUserDefinedString(entity.entityType)
+    validateUserDefinedString(entity.name)
+    entity.attributes.keys.foreach(validateUserDefinedString)
+
```

Comment:
I think this needs some testing. Add an entity called `dotted.name` and then go looking for it by name. I'm worried that Orient will parse it wrong.